### PR TITLE
Fix duplicate payments issue

### PR DESCRIPTION
**Pull request by LexMalta**
Strap in, because this one is a doozy to explain, but with such a simple fix.

JIRA: https://myxplorinfo.atlassian.net/browse/ZT-25994

## TLDR
An HTTP client timeout was too short, causing us to think payments had failed when they had in fact gone through successfully.
Lots of indirection, bad code, and mixed assumptions at various level in the codebase made it very hard to track down.

## Background Context
- Schools: Our Schools payment product, revolves around parents in a school being able to pay for various school-related things such as school fees, excursions, canteen, donations, etc etc
- Ironbank: The schools platform is made up of several "microservices". Ironbank is one of them and it is the service that deals with money and payments
- payway-elixir: this repo - a "stand alone" library used by Ironbank to provide a layer of abstraction in making API calls to Payway (Westpac's payment gateway).

One of the things Ironbank deals with is reoccurring payments. Parents can setup a reoccurring payment schedule (e.g. pay $100 every week), and Ironbank will do the work to actually make the payment at the appropriate time. Every 2 hours, on the hour, a cron job will go off in Ironbank that will check the scheduled payments, see if there are any that are due and make a payment if necessary.

## The problem
For the last few months Schools has been suffering from a highly intermittent duplicate payments issue. Parents with a reoccurring payment would occasionally, completely randomly, be double-charged - and Ironbank would only have a record of a single payment being in the DB. However when speaking to Payway support, it was clear that the issue was within Ironbank:

- The payments were clearly being made by our API keys
- The payments were made perfectly on the hour
- The second payment was always exactly 2 hours after the first one
- Ironbank would only ever have a record of the second payment in its db
- We could only find a log of the second API call made in Cloudwatch

## The cause
- When Ironbank processes a reoccurring payment it starts off with a call to `Payments.initiate` around: https://github.com/myxplor/ironbank/blob/master/lib/ironbank/services/schedules.ex#L112
- `Payments.initiate` will get called with a `strategy` of `Payway`, which will call `Payway.make_payment`: https://github.com/myxplor/ironbank/blob/325ded1d8cd9033dc7ab868172b4a295eca4d1e9/lib/ironbank/services/payments.ex#L28
- `Payway.make_payment` will call `Payway.Payments.new`: https://github.com/myxplor/ironbank/blob/master/lib/ironbank/services/strategies/payway.ex#L27
- `Payway.Payments.new` will eventually call `API.,Transaction.make_payment` - the API namespace being `payway-elixir` : https://github.com/myxplor/ironbank/blob/master/lib/ironbank/services/strategies/payway/payments.ex#L45

- `API.Transaction.new` will eventually call `Payway.post`: https://github.com/myxplor/payway-elixir/blob/master/lib/payway/api/transaction.ex#L40
- `Payway.post` will eventually calls `REST.post!`: https://github.com/myxplor/payway-elixir/blob/40a3c9e2fd876f35edf8b01260fbafaddbc24560/lib/payway.ex#L37
- `REST.post!` is an [HTTPoison.Base](https://hexdocs.pm/httpoison/0.13.0/HTTPoison.Base.html)
- `REST.post!` has a `recv_timeout` of 10 seconds and being a `post!` (with an `!`) it throws an exception if the API call "fails"
- The exception that gets thrown is handled all the way back right at the very top of this chain at: https://github.com/myxplor/ironbank/blob/master/lib/ironbank/services/schedules.ex#L122, and caught exceptions are logged into Appsignal
- [Here is an appsignal error](https://appsignal.com/myxplor/sites/5a94e60816b7e2341ed112e2/exceptions/incidents/73/samples/5a94e60816b7e2341ed112e2-854700277274370186916580592001?activeFilter=date&selectedTime=2022-07-16T14%3A00%3A00.000Z) for a duplicate payment that we do not have any record of in the db or any logs for in Cloudwatch - the timestamp of this log in Appsignal matches to-the-second the timestamp of the duplicate payment on Payway's end.
- You can see in the Appsignal log the cause of the error is HTTPoison timing out

So...HTTPoison thinks a request is timing out, but the payment is successfully in Payway. This means the client must be timing out before the actual request is complete. 
Kinda understandable, 10 seconds is actually a pretty short amount of time for a payment processor to respond to a request.

## The fix
- Increase the HTTP client-side timeout on API requests made from 10 seconds to 5 minutes
- Add logging to API calls made to Payway at the `REST` module level so we won't have unlogged calls made to Payway anymore
  - This required a slight rejigging of the module so I could log API call results _before_ an exception got thrown 
 - In order to compile payway-elixir stand-alone (not within Ironbank) I had to update some dependencies. I updated them to match the versions of the dependencies used in Ironbank.

### CODE REVIEW

**Review by adam-dicker — APPROVED** [review]
looks sane to me. Great work mate!

**Review by hjemmel — APPROVED** [review]
LGTM
